CodeGen/Generators/UnitsNetGen/QuantityGenerator.cs

```diff
@@ -1123,29 +1113,44 @@ private bool TryToUnit({_unitEnumName} unit, [NotNullWhen(true)] out {_quantity.
             converted = convertedOrNull.Value;
             return true;
         }}
-
-        /// <inheritdoc />
-        IQuantity IQuantity.ToUnit(Enum unit)
+");
+            Writer.WL(_isDimensionless
+                ? $@"
+        /// <inheritdoc cref=""IQuantity.ToUnit(UnitSystem)""/>
+        public {_quantity.Name} ToUnit(UnitSystem unitSystem)
         {{
-            if (!(unit is {_unitEnumName} typedUnit))
-                throw new ArgumentException($""The given unit is of type {{unit.GetType()}}. Only {{typeof({_unitEnumName})}} is supported."", nameof(unit));
+            if (unitSystem is null)
+                throw new ArgumentNullException(nameof(unitSystem));
 
-            return ToUnit(typedUnit, DefaultConversionFunctions);
+            return ToUnit(BaseUnit);
         }}
-
+"
+                : $@"
         /// <inheritdoc cref=""IQuantity.ToUnit(UnitSystem)""/>
         public {_quantity.Name} ToUnit(UnitSystem unitSystem)
         {{
-            if (unitSystem is null)
-                throw new ArgumentNullException(nameof(unitSystem));
+            return ToUnit(Info.GetDefaultUnit(unitSystem));
+        }}
+");
```

Comment:
Yes, I think it might actually be better in terms of consistency - I hadn't thought about what a potential call to `Ratio.Info.GetDefaultUnit(UnitSystem.SI)` would yield..

Reply:
Fixed in https://github.com/angularsen/UnitsNet/pull/1473/commits/4fc87b47ed74847fbb29eca5a936dab93107b4cc (I put it in the `QuantityTestsBase`, if you want a `TestConstants` class- pick the location)